### ISSUE TITLE
Multistep (predictions in) anomaly score

### ISSUE THREAD

**Issue by breznak** [numenta/nupic#1937]
the `MultiStep` models can make predictions `n-steps` in the future. 
(this is somewhat hacky in TP, cleaner in TM, as far as I understand..)

But for `Anomaly`? the doc for `computeRawAnomalyScore()` and code in `CLAModel` calling anomaly look like the score is computed just for T, T-1. Or does it get shifted somewhere? Because if not, results for anomaly models on multistep predictions (anomaly models train just like prediction models) would look better than in reality. 

@chetan51 ?

**Comment by rhyolight** [comment]
I never got the impression that the anomaly scores are tied to the N-step ahead prediction. I don't think it matters how many steps ahead the model is configured for.

**Comment by cogmission** [comment]
You are correct. There are two methods: the one-off computeRawAnomalyScore() which keeps no history, and the other method (can't remember now) which optionally includes a moving average over anomaly scores with an "adjustable" window - but there is no way to determine or configure n-steps ahead as the anomaly computer is only given a sample of the current distribution.

Unless I missed something?

**Comment by breznak** [comment]
@rhyolight it is not possible, but one might want to do that. @cogmission is right. 
* anomaly is based on comparing `current activeState` with `predictiveState from some time ago` generally 
* currently the "some time ago" is hard-coded to `T, T-1`
* like you can have a predictive model that makes `N-step ahead` predictions
* you could use these prediction to construct "N-step ahead" `anomaly model`
